src/main/java/io/cryostat/recordings/RecordingMetadataManager.java

```diff
@@ -780,17 +826,28 @@ private boolean targetRecordingExists(ConnectionDescriptor cd, String recordingN
         }
     }
 
-    private Path getMetadataPath(String jvmId, String recordingName) throws IOException {
+    private Path getMetadataPath(ConnectionDescriptor connectionDescriptor, String recordingName)
+            throws IOException {
+        String jvmId = jvmIdHelper.getJvmId(connectionDescriptor);
+        return getMetadataPath(connectionDescriptor.getTargetId(), jvmId, recordingName);
+    }
+
+    private Path getMetadataPath(String targetId, String jvmId, String recordingName)
+            throws IOException {
         String subdirectory =
                 jvmId.equals(UPLOADS)
                         ? UPLOADS
                         : base32.encodeAsString(jvmId.getBytes(StandardCharsets.UTF_8));
         String filename =
                 base32.encodeAsString(recordingName.getBytes(StandardCharsets.UTF_8)) + ".json";
-        if (!fs.exists(recordingMetadataDir.resolve(subdirectory))) {
-            fs.createDirectory(recordingMetadataDir.resolve(subdirectory));
+        Path subdirPath = recordingMetadataDir.resolve(subdirectory);
+        if (!fs.exists(subdirPath)) {
+            fs.createDirectory(subdirPath);
+            if (!jvmId.equals(UPLOADS)) {
+                fs.writeString(subdirPath.resolve("connectUrl"), targetId);
```

Comment:
I'm wondering why the metadata paths also need these connectUrl files?

Reply:
IIRC I needed to do this because I removed the external mutability of the `JvmIdHelper`. The old startup logic would read information from the filesystem and then during startup insert those values into the JVM ID map and transfer things between restarted instances that way. The new logic doesn't allow for any of that, and if we have a JVM ID computed at all then it must have come from the current instance. So the `connectUrl` is also recorded on the filesystem so that on startup we can read that and do the same transferring/correlation  as we do for the archived recordings files and re-attach metadata to archived recordings if the source target restarted but has the same targetId/connectUrl.